ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/RewardTraceGeneratorTest.java

```diff
@@ -90,7 +90,7 @@ public void assertThatTraceGeneratorReturnValidRewardsForMainnetBlockProcessor()
         new MainnetBlockProcessor(
             transactionProcessor,
             transactionReceiptFactory,
-            blockReward,
+            protocolSchedule,
```

Comment:
The ProtocolSchedule is effectively a list of ProtocolSpecs - each of which defines the behaviour for each block in a milestone.
As such - Besu creates a _new_ MainnetBlockProcessor for each milestone, each potentially with a different blockReward.
Thus, the goal is for a ProtocolSchedule to have a number of BlockProcessors, rather than a single BlockProcessor having access to the schedule.